### PR TITLE
doc:API.md/changes to bring content in sync with docs.minio.io

### PR DESCRIPTION
**Pull request by koolhead17**
changes to bring content in sync with docs.minio.io

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -1,445 +1,648 @@
...
+from minio import Minio
+from minio.error import ResponseError
+
+minioClient = Minio('play.minio.io:9000',
+                  access_key='Q3AM3UQ867SPQQA43P2F',
+                  secret_key='zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG',
+                  secure=True)
+```
+
+``2.AWS S3``
```
Comment (GarimaKapoor) [inline API.md]: Space between 2. and AWS

File: API.md [hunk 2]
```diff
@@ -1,445 +1,648 @@
...
-
-    # print curl command to upload files.
-    curl_cmd.append('-F file=@<FILE>')
-    print(' '.join(curl_cmd))
-```
+# Python Client API Reference
+
+Initialize Minio Client object.
+
+``1. Minio``
```
Comment (GarimaKapoor) [inline API.md]: Single space between 1. and Minio

File: API.md [hunk 3]
```diff
@@ -1,445 +1,648 @@
...
+## 1. Constructor
+---------------------------------------
+<a name="Minio">
+#### Minio(endpoint, access_key=None, secret_key=None, secure=True)
+
+|   |
+|---|
+| `Minio(endpoint, access_key=None, secret_key=None, secure=True)`  |
+| Initializes a new client object.  |
+
```
Comment (GarimaKapoor) [inline API.md]: Formatting of the table is not correct. There is additional row on the top.

File: API.md [hunk 4]
```diff
@@ -1,445 +1,648 @@
...
+```py
+from minio import Minio
+from minio.error import ResponseError
+
+minioClient = Minio('play.minio.io:9000',
+                    access_key='Q3AM3UQ867SPQQA43P2F',
+                    secret_key='zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG')
+```
+
+``2.AWS S3``
```
Comment (GarimaKapoor) [inline API.md]: spacing is not correct between 2. and AWS

File: API.md [hunk 5]
```diff
@@ -1,445 +1,648 @@
...
+from minio import Minio
+from minio.error import ResponseError
+
+s3Client = Minio('s3.amazonaws.com',
+                 access_key='ACCESS_KEY',
+                 secret_key='SECRET_KEY')
+```
+
+## 2. Bucket operations
+---------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: The line under the bucket operations should be removed.

**Comment by harshavardhana** [comment]
> Formatting of the table is not correct. There is additional row on the top.

Let me fix these on the master, now i see the format.

File: API.md [hunk 6]
```diff
@@ -1,445 +1,648 @@
...
+            </tr>
+            <tr>
+            <td>location</td>
+            <td>string</td>
+            <td>Default value is us-east-1<br/>
+Location valid values are us-west-1, us-west-2, eu-west-1, eu-central-1, ap-southeast-1, ap-northeast-1, ap-southeast-2, sa-east-1(defaults to us-east-1, optional).</td>
+            </tr>
+               </tbody>
+</table>
+
```
Comment (GarimaKapoor) [inline API.md]: Only the parameters should be highlighted and not the type or description.

Comment (GarimaKapoor) [inline API.md]: Type needs to be italics. if it is not possible here then bring other types to same formatting.

File: API.md [hunk 7]
```diff
@@ -1,445 +1,648 @@
...
+__Example__
+```py
+try:
+    minioClient.make_bucket("mybucket", location="us-east-1")
+except ResponseError as err:
+    print(err)
+```
+
+---------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: The additional line at the end is not required.

File: API.md [hunk 8]
```diff
@@ -1,445 +1,648 @@
...
+|``bucket_name``   |*string*   |Bucket name. |
+|``bucket.creation_date`` |*time*   |Time: date when bucket was created. |
+
+__Example__
+```py
+buckets = minioClient.list_buckets()
+for bucket in buckets:
+    print(bucket.name, bucket.creation_date)
+```
+---------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Additional line 143 is not required.

File: API.md [hunk 9]
```diff
@@ -1,445 +1,648 @@
...
+
+__Example__
+
+```py
+try:
+    print(minioClient.bucket_exists("mybucket"))
+except ResponseError as err:
+    print(err)
+```
+---------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 162. Additional line is not required.

File: API.md [hunk 10]
```diff
@@ -1,445 +1,648 @@
...
+|``bucket_name``   |*string*   |Name of the bucket. |
+
+__Example__
+```py
+try:
+    minioClient.remove_bucket("mybucket")
+except ResponseError as err:
+    print(err)
+```
+---------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 180. Not required

File: API.md [hunk 11]
```diff
@@ -1,445 +1,648 @@
...
+            <li>object.object_name string: name of the object. </li>
+            <li>object.size int: size of the object.</li>
+            <li>object.etag string: etag of the object. </li>
+            <li>object.last_modified datetime.datetime: modified time stamp. </li>
+            </ul>
+            </td>
+            </tr>
+               </tbody>
+</table>
+
```
Comment (GarimaKapoor) [inline API.md]: Return Value parameter needs to be highlighted and the type needs to be in italics.

File: API.md [hunk 12]
```diff
@@ -1,445 +1,648 @@
...
+```py
+# List all object paths in bucket that begin with my-prefixname.
+objects = minioClient.list_objects('mybucket', prefix='my-prefixname',
+                              recursive=True)
+for obj in objects:
+    print(obj.bucket_name, obj.object_name.encode('utf-8'), obj.last_modified,
+          obj.etag, obj.size, obj.content_type)
+```
+--------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 231. Not required.

File: API.md [hunk 13]
```diff
@@ -1,445 +1,648 @@
...
+Lists partially uploaded objects in a bucket.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucketname``   | *string*  |Name of the bucket.|
+|``prefix``   |*string*    |The prefix of the incomplete objects uploaded should be listed. |
+|``recursive`` |*bool*   |``True`` indicates recursive style listing and ``False`` indicates directory style listing delimited by '/'. Optional default is ``False``.   |
+
```
Comment (GarimaKapoor) [inline API.md]: Second line in the parameter table should not highlighted fully. Only the params need to be highlighted.

File: API.md [hunk 14]
```diff
@@ -1,445 +1,648 @@
...
+            <ul>
+            <li>multipart_obj.object_name string: name of the incomplete object.</li>
+            <li>multipart_obj.upload_id string: upload ID of the incomplete object.</li>
+            <li>multipart_obj.size int: size of the incompletely uploaded object. </li>
+            </ul>
+            </td>
+            </tr>
+               </tbody>
+</table>
+
```
Comment (GarimaKapoor) [inline API.md]: Return value: Param need to be highlighted and the type should be in italics. The different iterators in the description should be highlighted. Refer to https://docs.minio.io/docs/python-client-api-reference for the correct formatting.

File: API.md [hunk 15]
```diff
@@ -1,445 +1,648 @@
...
+```py
+# List all object paths in bucket that begin with my-prefixname.
+uploads = minioClient.list_incomplete_uploads('mybucket',
+                                         prefix='my-prefixname',
+                                         recursive=True)
+for obj in uploads:
+    print(obj.bucket_name, obj.object_name, obj.upload_id, obj.size)
+```
+--------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 281- not required.

File: API.md [hunk 16]
```diff
@@ -1,445 +1,648 @@
...
+<a name="get_object">
+#### get_object(bucket_name, object_name)
+Downloads an object.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*   |Name of the object.   |
```
Comment (GarimaKapoor) [inline API.md]: The second line in the parameters table should not be highlighted.

File: API.md [hunk 17]
```diff
@@ -1,445 +1,648 @@
...
+# Get a full object.
+try:
+    data = minioClient.get_object('mybucket', 'myobject')
+    with open('my-testfile', 'wb') as file_data:
+        for d in data:
+            file_data.write(d)
+except ResponseError as err:
+    print(err)
+```
+--------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 311: not required.

File: API.md [hunk 18]
```diff
@@ -1,445 +1,648 @@
...
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+|``offset``   |*int*   |``offset`` of the object from where the stream will start.   |
+|``length``   |*int*    |``length`` of the object that will be read in the stream (optional, if not specified we read the rest of the file from the offset).   |
+
```
Comment (GarimaKapoor) [inline API.md]: Second and the fourth line of the table are greyed out. Need to be fixed

File: API.md [hunk 19]
```diff
@@ -1,445 +1,648 @@
...
+    print(err)
+```
+--------------------------------------
+<a name="get_partial_object">
+#### get_partial_object(bucket_name, object_name, offset=0, length=0)
+Downloads the specified range bytes of an object.
+
+__Parameters__
+
+|Param   |Type   |Description   |
```
Comment (GarimaKapoor) [inline API.md]: Description title should be left aligned in the table instead of center.

File: API.md [hunk 20]
```diff
@@ -1,445 +1,648 @@
...
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+|``offset``   |*int*   |``offset`` of the object from where the stream will start.   |
+|``length``   |*int*    |``length`` of the object that will be read in the stream (optional, if not specified we read the rest of the file from the offset).   |
+
+__Return Value__
+
+|Param   |Type   |Description   |
```
Comment (GarimaKapoor) [inline API.md]: Description  should be left aligned.

File: API.md [hunk 21]
```diff
@@ -1,445 +1,648 @@
...
+#### fget_object(bucket_name, object_name, file_path)
+Downloads and saves the object as a file in the local filesystem.
+
+__Parameters__
+
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
```
Comment (GarimaKapoor) [inline API.md]: This line should not be greyed out.

File: API.md [hunk 22]
```diff
@@ -1,445 +1,648 @@
...
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+|``data``   |*io.IOBase*   |Any python object implementing io.IOBase. |
+|``length``   |*int*   |Total length of object.   |
+|``content_type``   |*string* | Content type of the object. (optional, defaults to 'application/octet-stream').   |
+
```
Comment (GarimaKapoor) [inline API.md]: Second and fourth line should not be greyed out. Needs to be fixed

File: API.md [hunk 23]
```diff
@@ -1,445 +1,648 @@
...
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+|``file_path``   |*string*   |Path on the local filesystem to which the object data will be written. |
+|``content_type``   |*string* | Content type of the object. (optional, defaults to 'application/octet-stream').   |
+
```
Comment (GarimaKapoor) [inline API.md]: Second and fourth line should not be greyed out.

File: API.md [hunk 24]
```diff
@@ -1,445 +1,648 @@
...
+# Put on object 'myobject.csv' with contents from
+# '/tmp/otherobject.csv' as 'application/csv'.
+try:
+    minioClient.fput_object('mybucket', 'myobject.csv',
+                       '/tmp/otherobject.csv', content_type='application/csv')
+except ResponseError as err:
+    print(err)
+
+```
+------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 439: Not required

File: API.md [hunk 25]
```diff
@@ -1,445 +1,648 @@
...
+<a name="stat_object">
+#### stat_object(bucket_name, object_name)
+Gets metadata of an object.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
```
Comment (GarimaKapoor) [inline API.md]: The second line of the table is greyed out. Needs to be fixed.

File: API.md [hunk 26]
```diff
@@ -1,445 +1,648 @@
...
+            <li>obj.size int: size of the object.</li>
+            <li>obj.etag string: etag of the object.</li>
+            <li>obj.content_type string: Content-Type of the object.</li>
+            <li>obj.last_modified time.time: modified time stamp.</li>
+            </ul>
+            </td>
+            </tr>
+               </tbody>
+</table>
+
```
Comment (GarimaKapoor) [inline API.md]: Param needs to be highlighted. Type needs to in italics. Object stat under description needs to be highlighted.

File: API.md [hunk 27]
```diff
@@ -1,445 +1,648 @@
...
+__Example__
+```py
+# Fetch stats on your object.
+try:
+    print(minioClient.stat_object('mybucket', 'myobject'))
+except ResponseError as err:
+    print(err)
+```
+------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Line 487 should be removed: not required.

File: API.md [hunk 28]
```diff
@@ -1,445 +1,648 @@
...
+#### remove_object(bucket_name, object_name)
+Removes an object.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+
```
Comment (GarimaKapoor) [inline API.md]: Second line in the table should not be greyed out. Needs to be fixed.

File: API.md [hunk 29]
```diff
@@ -1,445 +1,648 @@
...
+```py
+# Remove an object.
+try:
+    minioClient.remove_object('mybucket', 'myobject')
+except ResponseError as err:
+    print(err)
+```
+
+-----------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Line 509 needs to be removed: not required

File: API.md [hunk 30]
```diff
@@ -1,445 +1,648 @@
...
+#### remove_incomplete_upload(bucket_name, object_name)
+Removes a partially uploaded object.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+
```
Comment (GarimaKapoor) [inline API.md]: Second line should not be greyed out.

File: API.md [hunk 31]
```diff
@@ -1,445 +1,648 @@
...
+```py
+# Remove an partially uploaded object.
+try:
+    minioClient.remove_incomplete_upload('mybucket', 'myobject')
+except ResponseError as err:
+    print(err)
+```
+## 4. Presigned operations
+---------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 531: Not Required

File: API.md [hunk 32]
```diff
@@ -1,445 +1,648 @@
...
+Generates a presigned URL for HTTP GET operations. Browsers/Mobile clients may point to this URL to directly download objects even if the bucket is private. This presigned URL can have an associated expiration time in seconds after which it is no longer operational. The default expiry is set to 7 days.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+|``expiry``   | *datetime.datetime*    |Expiry in seconds. Default expiry is set to 7 days.    |
+
```
Comment (GarimaKapoor) [inline API.md]: Second row of the table is greyed out: needs to be fixed.

File: API.md [hunk 33]
```diff
@@ -1,445 +1,648 @@
...
+# presigned get object URL for object name, expires in 2 days.
+try:
+    print(minioClient.presigned_get_object('mybucket', 'myobject', expires=timedelta(days=2)))
+# Response error is still possible since internally presigned does get bucket location.
+except ResponseError as err:
+    print(err)
+```
+
+------------------------------------
```
Comment (GarimaKapoor) [inline API.md]: Remove line 557: Not required

File: API.md [hunk 34]
```diff
@@ -1,445 +1,648 @@
...
+NOTE: you can upload to S3 only with specified object name.
+
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|---|---|---|
+|``bucket_name``   |*string*   |Name of the bucket.   |
+|``object_name``   |*string*    |Name of the object.   |
+|``expiry``   | *datetime.datetime*    |Expiry in seconds. Default expiry is set to 7 days.    |
```
Comment (GarimaKapoor) [inline API.md]: Second row of the table should not be greyed out. Needs to be fixed

File: API.md (hunk 2 above)
Comment (koolhead17) [inline API.md]: done.

File: API.md (hunk 1 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 3 above)
Comment (koolhead17) [inline API.md]: as per discussion will talk to @rushenn for this fix.

File: API.md (hunk 4 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 5 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 6 above)
Comment (koolhead17) [inline API.md]: will be working with @rushenn on same.

File: API.md (hunk 7 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 8 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 9 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 10 above)
Comment (koolhead17) [inline API.md]: done.

File: API.md (hunk 11 above)
Comment (koolhead17) [inline API.md]: working on fix so we have same theme for html table as well as markdown with @rushenn

File: API.md (hunk 12 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 13 above)
Comment (koolhead17) [inline API.md]: Github default theme, will work with @rushenn to get solution for same.

File: API.md (hunk 15 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 16 above)
Comment (koolhead17) [inline API.md]: default github theme 😢

File: API.md (hunk 17 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 19 above)
Comment (koolhead17) [inline API.md]: work with @rushenn on it.

File: API.md (hunk 21 above)
Comment (koolhead17) [inline API.md]: github default theme

File: API.md (hunk 24 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 27 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 29 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 31 above)
Comment (koolhead17) [inline API.md]: done

File: API.md (hunk 33 above)
Comment (koolhead17) [inline API.md]: done

**Pull request merged**